### PR TITLE
PAINTROID-458: Fix multi-touch brush tool behaviour

### PR DESCRIPTION
**Pull request by davidmoling**
Ticket: https://jira.catrob.at/browse/PAINTROID-458

### CODE REVIEW

File: lib/workspace/src/ui/drawing_canvas.dart [hunk 1]
```diff
@@ -29,20 +31,29 @@ class _DrawingCanvasState extends ConsumerState<DrawingCanvas>
         ),
         position: DecorationPosition.foreground,
         child: GestureDetector(
-          onPanDown: (details) {
+          onScaleStart: (details) {
             final box = context.findRenderObject() as RenderBox;
-            toolStateNotifier
-                .didTapDown(box.globalToLocal(details.globalPosition));
+            multiScaleInProgress = true; // prevent first dot draw bug
```
Comment (CodeChamp-SS) [inline lib/workspace/src/ui/drawing_canvas.dart]: is this `multiScaleInProgress` variable required? Can't check directly with the if condition in `onScaleEnd`? 🤔

File: lib/workspace/src/ui/drawing_canvas.dart [hunk 2]
```diff
@@ -29,20 +31,29 @@ class _DrawingCanvasState extends ConsumerState<DrawingCanvas>
...
+              canvasDirtyNotifier.repaint();
+            } else {
+              multiScaleInProgress = true;
+            }
           },
-          onPanEnd: (_) {
-            toolStateNotifier.didTapUp();
-            canvasStateNotifier.updateCachedImage();
+          onScaleEnd: (details) {
+            if (!multiScaleInProgress) {
```
Comment (CodeChamp-SS) [inline lib/workspace/src/ui/drawing_canvas.dart]:
```suggestion
            if (details.pointerCount == 1) {
```

**Review by CodeChamp-SS — CHANGES_REQUESTED** [review]

File: lib/workspace/src/ui/drawing_canvas.dart (hunk 1 above)
Comment (davidmoling) [reply lib/workspace/src/ui/drawing_canvas.dart:36]: yes, it was tried without, but then a first dot is drawn when scaling. This line disables drawing until it is determined for sure that it is a single touch.

Comment (davidmoling) [reply lib/workspace/src/ui/drawing_canvas.dart]: does not work unfortunately with the suggested way, a single dot is drawn on scale

File: lib/workspace/src/ui/drawing_canvas.dart (hunk 2 above)
Comment (davidmoling) [reply lib/workspace/src/ui/drawing_canvas.dart]: see comment above

File: lib/workspace/src/ui/drawing_canvas.dart (hunk 1 above)
Comment (CodeChamp-SS) [reply lib/workspace/src/ui/drawing_canvas.dart]: oh ok

**Review by CodeChamp-SS — APPROVED** [review]
LGTM

**Pull request merged**